src/python/gudhi/cubical_complex.pyx

```diff
@@ -47,7 +47,7 @@ cdef class CubicalComplex:
 
     # Fake constructor that does nothing but documenting the constructor
     def __init__(self, dimensions=None, top_dimensional_cells=None,
-                  perseus_file=''):
+                 numpy_array=None, perseus_file=''):
```

Comment:
I don't like the name `numpy_array`. But it is true that top_dimensional_cells is taken, and we want something that plays the same role. How about this: we keep the same parameters as before, but if `dimensions` is None, top_dimensional_cells is expected to be a numpy array and we deduce the dimensions from its shape. Does that make sense?

Reply:
I am ok with your proposal. I will try to make it clear in the documentation.

Reply:
I am ok with your proposition.
The fix is done on b63be266effe24646823acc59fe397021bb13a3e